vertical-pod-autoscaler/pkg/updater/logic/updater_test.go

```diff
@@ -357,3 +359,50 @@ func TestRunOnceIgnoreNamespaceMatching(t *testing.T) {
 	updater.RunOnce(context.Background())
 	eviction.AssertNumberOfCalls(t, "Evict", 0)
 }
+
+func TestNewEventRecorder(t *testing.T) {
+	fakeClient := fake.NewSimpleClientset()
+	er := newEventRecorder(fakeClient)
+
+	testCases := []struct {
+		reason  string
+		object  runtime.Object
+		message string
+	}{
+		{
+			reason:  "EvictedPod",
+			object:  &apiv1.Pod{},
+			message: "Evicted pod",
+		},
+		{
+			reason:  "EvictedPod",
+			object:  &vpa_types.VerticalPodAutoscaler{},
+			message: "Evicted pod",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.reason, func(t *testing.T) {
+			er.Event(tc.object, apiv1.EventTypeNormal, tc.reason, tc.message)
+
+			var events *apiv1.EventList
+			var err error
+			// Add delay for fake client to catch up due to be being asynchronous
+			for i := 0; i < 5; i++ {
+				events, err = fakeClient.CoreV1().Events("default").List(context.TODO(), metav1.ListOptions{})
```

Comment:
Instead of `context.TODO()` I suggest something like:
```
ctx, cancel := context.WithTimeout(context.Background(), contextTimeout)
defer cancel()
```

Reply:
Great, thanks. Done in https://github.com/kubernetes/autoscaler/pull/7413/commits/d16c5e7aaf0f8c2346b2350d244aba75c67653ac